MrtCore/console_unpackaged_app/console_unpackaged_app.cpp

```diff
@@ -0,0 +1,114 @@
+            "  Default and override retrieve a sample string from string resource files.\n"
+            "  For the override case, this sample uses the German language.\n"
+            "  Fallback corresponds to the resource-not-found case, where we fallback to a legacy resource loader.\n"
+            "\n"
+            "Examples:\n"
+            "  Get the sample string for the default resource context\n"
+            "    console_unpackaged_app.exe default\n"
+            "  Get the sample string for the override resource context (sample uses the German language for the override context)\n"
+            "    console_unpackaged_app.exe override\n"
+            "  Get the sample string for the resource-not-found fallback case\n"
+            "    console_unpackaged_app.exe fallback\n";
+        return 1;
+    }
+
+    // Initialize dynamic dependencies so we can consume the Project Reunion APIs in the Project Reunion framework package from this unpackaged app. 
+    HRESULT mddInitHr = MddBootstrapInitializeWrapper();
+    if (FAILED(mddInitHr))
+    {
+        return 1;
+    }
+
+    ResourceManager manager = InitMrtCore();
+    manager.ResourceNotFound([](ResourceManager const&, ResourceNotFoundEventArgs const& args)
+        {
+            // There could be a resource in a legacy resource file that we retrieve using the corresponding legacy resource loader. For example, the C#
+            // MRT Core sample apps in this repo use the .NET resource loader as their fallback. Instead, we just hardcode a resource value here.
+            if (args.Name() == L"Resources/LegacyString")
+            {
+                auto candidate = ResourceCandidate(ResourceCandidateKind::String, L"Legacy Sample");
+                args.SetResolvedCandidate(candidate);
+            }
+        });
+
+    if ((wcscmp(argv[1], L"Default") == 0) || (wcscmp(argv[1], L"default") == 0))
+    {
+        std::wcout << manager.MainResourceMap().GetValue(L"Resources/SampleString").ValueAsString().c_str() << std::endl;
+    }
+    else if ((wcscmp(argv[1], L"Override") == 0) || wcscmp(argv[1], L"override") == 0)
+    {
+        // Create a custom resource context. Set the language to German.
+        ResourceContext overrideResourceContext = manager.CreateResourceContext();
+        overrideResourceContext.QualifierValues().Insert(L"Language", L"de-DE");
+
+        // Use the custom resource context for the resource lookup.
+        std::wcout << manager.MainResourceMap().GetValue(L"Resources/SampleString", overrideResourceContext).ValueAsString().c_str() << std::endl;
+    }
+    else if ((wcscmp(argv[1], L"Fallback") == 0) || (wcscmp(argv[1], L"fallback") == 0))
+    {
+        std::wcout << manager.MainResourceMap().GetValue(L"Resources/LegacyString").ValueAsString().c_str() << std::endl;
+    }
+    else
+    {
+        std::wcout << "Invalid argument!" << std::endl;
+    }
+
+    // Uninitialize dynamic dependencies.
+    MddBootstrapShutdown();

```

Comment:
Consider adding a wil-based helper that does this for you.

Reply:
@jonwis Jon, would you please suggest a wil RAII type I can use here?

Reply:
I think you could maybe use `wil::scope_exit` for calling this:

https://github.com/microsoft/wil/wiki/RAII-resource-wrappers#wilscope_exit

```c++
auto cleanup = wil::scope_exit([&]
{
    MddBootstrapShutdown();
});
```